shuup/utils/money.py

```diff
@@ -60,3 +109,44 @@ def unit_matches_with(self, other):
 
     def new(self, value):
         return type(self)(value, currency=self.currency)
+
+    @classmethod
+    def set_precision_provider_class(cls, precision_provider_class):
+        """
+        Set the precision provider class to be used when rounding
+
+        If some precision provider is currently instantiated,
+        it will be destroyed.
+
+        :param precision_provider_class type
+        """
+        assert issubclass(precision_provider_class, BasePrecisionProvider)
+        cls.precision_provider_class = precision_provider_class
+
+        # clear the current precision provider instance
+        cls.precision_provider = None
+
+    def as_rounded(self, digits=None, rounding=ROUND_HALF_EVEN):
+        """
+        Returns the value rounded to the given `digits` if specified,
+        otherwise through the currency of the this instance.
+
+        The precision will be fetched from the `precision_provider`.
+
+        :param digits int:
+          The number of digits to round
+        :type rounding: str
+
+        :return A new value rounded to the given digits or currency
+        """
+        cls = type(self)
+
+        if not cls.precision_provider:
+            cls.precision_provider = cls.precision_provider_class()
```

Comment:
Checking precision provider (and lazy initializing it) on a call to `as_rounded` doesn't feel like a good idea.